monitor/script.js

```diff
@@ -0,0 +1,246 @@
+        .join(
+          "_"
+        )}_pending: indexingStatusForPendingVersion(subgraphName: "messari/${name}") {
+            subgraph
+            node
+            synced
+            health
+            fatalError {
+            message
+            handler
+            }
+            nonFatalErrors {
+            message
+            handler
+            }
+            chains {
+            network
+            chainHeadBlock {
+                number
+            }
+            earliestBlock {
+                number
+            }
+            latestBlock {
+                number
+            }
+            lastHealthyBlock {
+                number
+            }
+            }
+            entityCount
+        }
+    `;
+    if (fullCurrentQueryStr[fullCurrentQueryStr.length - 1].length > 80000) {
+      fullCurrentQueryStr[fullCurrentQueryStr.length - 1] += "}";
+      fullCurrentQueryStr.push(" query Status {");
+    }
+    if (fullPendingQueryStr[fullPendingQueryStr.length - 1].length > 80000) {
+      fullPendingQueryStr[fullPendingQueryStr.length - 1] += "}";
+      fullPendingQueryStr.push(" query Status {");
+    }
+  });
+  fullCurrentQueryStr[fullCurrentQueryStr.length - 1] += "}";
+  fullPendingQueryStr[fullPendingQueryStr.length - 1] += "}";
+
+  const currentStateDepos = deployments;
+  Object.keys(currentStateDepos).forEach((name) => {
+    deployments[name + "-pending"] = { ...deployments[name] };
+  });
+
+  const indexingStatusQueries = [
+    ...fullCurrentQueryStr,
+    ...fullPendingQueryStr,
+  ].map((query) =>
+    axios.post("https://api.thegraph.com/index-node/graphql", { query: query })
+  );
+  let indexData = [];
+  await Promise.all(indexingStatusQueries)
+    .then(
+      (response) =>
+      (indexData = response.map(
+        (resultData) => ({ data: resultData.data.data }.data)
+      ))
+    )
+    .catch((err) => console.log(err));
+  let tempCombinedIndexData = {};
+  indexData.forEach((x) => {
+    tempCombinedIndexData = { ...tempCombinedIndexData, ...x };
+  });
+  indexData = tempCombinedIndexData;
+  // invalidDeployments array holds the namestrings of all deployments that have failed indexing. These will not attempt further validation
+  const invalidDeployments = [];
+  Object.keys(indexData).forEach((indexDataName) => {
+    const realNameString = indexDataName.split("_").join("-");
+    if (indexDataName.includes("pending")) {
+      deployments[realNameString].url =
+        "https://api.thegraph.com/subgraphs/id/" +
+        indexData[indexDataName]?.subgraph;
+    }
+
+    let indexedPercentage = ((indexData[indexDataName]?.chains[0]?.latestBlock?.number - indexData[indexDataName]?.chains[0]?.earliestBlock?.number) / (indexData[indexDataName]?.chains[0]?.chainHeadBlock?.number - indexData[indexDataName]?.chains[0]?.earliestBlock?.number)) || 0;
+    indexedPercentage = indexedPercentage * 100;
+    if (indexedPercentage > 99.5) {
+      indexedPercentage = 100;
+    }
```

Comment:
why do we do this reassignment?

Reply:
When subgraphs are fully indexed, sometimes the number of the current block being indexed is a few numbers off of the chain head block, then catches up. Its best to round up the ones within .5% of 100% since they are likely synced to head but the block indexed can lag temporarily.